### PR TITLE
Ports: Don't set previously unset environment variables

### PR DESCRIPTION
**Pull request by fdellwing**
CC @timschumi

### CODE REVIEW

**Review by timschumi — CHANGES_REQUESTED** [review]
That's not what I imagined what you meant when saying that it leads to a lot of code duplication. :^)

Couldn't we do something like this:

```sh
environment_variables=(
    "SERENITY_STRIPPED_ENV=1"
    "HOME=${HOME}"
    "USER=${USER}"
    "TERM=${TERM}"
    "PATH=${PATH}"
)

# Repeat for all the other options that might be optional.
# Note: -n doesn't mean unset.
if [ -n "${EDITOR}" ]; then
    environment_variables+=("EDITOR=${EDITOR}")
fi

exec env -i "${environment_variables[@]}" "${@}"
```

**Comment by gmta** [comment]
You can factor out the code duplication and check if the variable is actually set by using something like:

```bash
keep_environment_variables=(
    'EDITOR'
    'IN_SERENITY_PORT_DEV'
    'MAKEJOBS'
    'SERENITY_ARCH'
    'SERENITY_TOOLCHAIN'
)

for environment_variable_name in "${keep_environment_variables[@]}"; do
    if [ -v "${environment_variable_name}" ]; then
        environment_variables+=("${environment_variable_name}=${!environment_variable_name}")
    fi
done
```

File: Ports/.strip_env.sh [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-    SERENITY_ARCH="${SERENITY_ARCH:-}" \
-    SERENITY_TOOLCHAIN="${SERENITY_TOOLCHAIN:-}" \
-    "${@}"
+environment_variables=(
+    "SERENITY_STRIPPED_ENV=1"
+    "HOME=${HOME}"
+    "USER=${USER}"
+    "TERM=${TERM}"
+    "PATH=${PATH}"
+)
```
Comment (timschumi) [inline Ports/.strip_env.sh:9]: While we are touching this anyways...

```suggestion
environment_variables=(
    "HOME=${HOME}"
    "PATH=${PATH}"
    "SERENITY_STRIPPED_ENV=1"
    "TERM=${TERM}"
    "USER=${USER}"
)
```

**Review by timschumi — CHANGES_REQUESTED** [review]
One last small nitpick.

**Review by timschumi — APPROVED** [review]

**Pull request merged**